allennlp_models/generation/dataset_readers/copynet_seq2seq.py

```diff
@@ -72,6 +77,15 @@ class CopyNetDatasetReader(DatasetReader):
     source_token_indexers : `Dict[str, TokenIndexer]`, optional
         Indexers used to define input (source side) token representations. Defaults to
         `{"tokens": SingleIdTokenIndexer()}`.
+    add_start_and_end_tokens : `bool`, optional (default = `True`)
+        The `CopyNet` model assumes that the source sequence starts with a special
```

Comment:
Is this true?  I don't think it is.  You removed the index lookups from the model, and nothing broke.  These tokens were entirely unused, and if the user wants to have some additional special tokens like this on the start side for whatever modeling reason they want, they are easy to add in the tokenizer configuration.  I would just remove this parameter.

Reply:
The model does make this assumption since adding special start/end tokens is a typical thing to do and the model needs to avoid computing "copy" scores for those special tokens so that there's no possibility of generating those tokens in a predicted target sequence.

Reply:
"needs to" is probably too strong to say, but I think that was the rational.

Reply:
But the model doesn't actually do that.  `self._src_start_index` was never used in the model.

Reply:
See here:
- https://github.com/allenai/allennlp-models/blob/master/allennlp_models/generation/models/copynet_seq2seq.py#L340
- https://github.com/allenai/allennlp-models/blob/master/allennlp_models/generation/models/copynet_seq2seq.py#L356

The encodings of the start and end tokens are sliced off in various places.

Reply:
Thanks, you're right, I missed that.  That's pretty brittle, though.  Might be nice to have that configurable somehow.  But probably better done in a separate PR.  For this one, I might consider at least leaving a note somewhere about how we always take off the first and last token.

Reply:
Agreed. I'll follow up with a PR for that